src/main/flight/interpolated_setpoint.h

```diff
@@ -0,0 +1,29 @@
+/*
+ * This file is part of Cleanflight and Betaflight.
+ *
+ * Cleanflight and Betaflight are free software. You can redistribute
+ * this software and/or modify this software under the terms of the
+ * GNU General Public License as published by the Free Software
+ * Foundation, either version 3 of the License, or (at your option)
+ * any later version.
+ *
+ * Cleanflight and Betaflight are distributed in the hope that they
+ * will be useful, but WITHOUT ANY WARRANTY; without even the implied
+ * warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this software.
+ *
+ * If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+#include "pid.h"
+#include "common/axis.h"
+#include <stdint.h>
+
+void interpolatedSpInit(const pidProfile_t *pidProfile);
+float interpolatedSpApply(int axis, float pidFrequency);
+float applyFFLimit(int axis, float value, float Kp, float currentPidSetpoint);
```

Comment:
We shouldn't be calling `applyFFLimit()` if the code is built without `USE_INTERPOLATED_SP`. Especially as combined with the `FAST_CODE_NOINLINE` this will result in an empty function call for no reason.

Reply:
Agreed, although please see my open question related to `ff_max_rate`. It might worth being considered applying a maximum setpoint rate based limitation for existing logic as well. I've yet to try it out in practice but it is something would make sense for both logic. WDYT?

Reply:
Potentially. But even then, splitting it the way you are doing it is going to lead to inefficient code in the critical path, so it should be done differently. (N.b.: `FAST_CODE_NOINLINE` does **not** put the function into ITCM RAM, it just prevents it from being inlined.)

Reply:
I've changed it so it now branches off in `pid.c` instead of potentially ending up in an empty method call.